Tests/MessagingPush/PushHandling/AutomaticPushClickHandlingIntegrationTests.swift

```diff
@@ -0,0 +1,214 @@
+    }
+
+    // MARK: local push notification support
+
+    func test_givenMultiplePushClickHandlers_expectOtherClickHandlerHandlesPush_expectInfiniteLoopPrevented() {
+        setupAutomaticPushClickHandlingFeature()
+
+        notificationCenterMock.currentDelegate = UNUserNotificationCenterDelegateMock()
+        otherPushClickDelegate.userNotificationCenterDidReceiveCalled = { _ in
+            // handle push click
+        }
+
+        let givenLocalNotification = getLocalNotification(pushId: "hard-coded-non-unique-id")
+
+        clickOnPushNotification(givenLocalNotification)
+
+        XCTAssertEqusl(pushClickHandlerMock.pushClickedCount, 1) // should only be handled 1 time
+
+        clickOnPushNotification(givenLocalNotification)
+
+        XCTAssertEqusl(pushClickHandlerMock.pushClickedCount, 2) // should be handled again, even though we have already handled push with that push ID before.
+    }
+    
+    // TODO: the test class here tests only when a push is clicked. Add test functions for when a push is received on the device and we need to determine if we should show the push while app in foreground or not. 
+    // Another way to put it, we need to test our SDK's logic of: 
+    // userNotificationCenter(willPresent:)
+}
+
+extension AutomaticPushClickHandlingInsetegrationTest {
+    func setupAutomaticPushClickHandlingFeature() {
+        // TODO: we might need to change this function body because we may not be able to run the swizzling code in the test function. Running this line below will setup the swizzling code.
+        MessagingPush.initialize()
+    }
+
+    func clickOnPushNotification(_ pushContent: [AnyHashable: Any]) {
+        // We are acting as if we are the iOS OS and a push notification got clicked on the device.
+
+        // First, iOS will get the current UNUserNotificationCenterDelegate instance set on the host iOS app.
+        let hostAppDelegate = notificationCenterMock.currentDelegate
+
+        // Then, iOS will call this function on the delegate.
+        // Note: It's important that we test that the `withContentHandler` callback function gets called either by our SDK (when we handle it), or the 3rd party handler.
+        //       We add an expectation to verify that 1 push click handler calls it.
+        let expectCompletionHandlerCalled = expectation(description: "Expect completion handler called by a click handler")
+        expectCompletionHandlerCalled.expectedFulfillmentCount = 1 // Test will fail if called 2+ times which could indicate a bug because only 1 push click handler should be calling it.
+
+        // TODO: We may need to change the syntax of this line below.
+        // It's difficult to create an instance of UNNotificationRequest because it's initializer is internal.
+        // Therefore, we may need to call a different function then the one below. However, we will make sure that we are still testing the logic of our SDK when this function is called in production.
```

Comment:
Bringing attention to this important note for what to expect in future PRs on the implementation of this project.